### DISCUSSION TITLE
Is there a max body size for the interceptor? If so, can we modify it?

### DISCUSSION THREAD

**Discussion by josegonzalez** [kedacore/http-add-on#1202] [Q&A]
As the post says. I have a use-case where it is useful to post very large bodies to the service in order to process a request (image uploads and that sort of thing). I want to get ahead of the problem by validating what the limits are in the interceptor and see if we can modify them. Happy to dig into the code to contribute features that allow modification.

**Comment by wozniakjan** [comment]
afaict there is probably no explicit limit imposed by the interceptor. It uses golang core net/http package to implement the reverse proxying in go here:
https://github.com/kedify/http-add-on/blob/45f645f19e6e76ede4b61a5278665ac92d1e576e/interceptor/handler/upstream.go
which configures the http transport here:
https://github.com/kedify/http-add-on/blob/45f645f19e6e76ede4b61a5278665ac92d1e576e/interceptor/proxy_handlers.go#L54-L64
what I was able to find is that in go,  http.MaxBytesReader is a way to limit the size of incoming traffic
https://pkg.go.dev/net/http#MaxBytesReader
and given it's not used anywhere in the interceptor codebase, it's possible there is no limit at all?
